prover/src/constraints/evaluator/default.rs

```diff
@@ -130,19 +150,29 @@ where
         // evaluations
         let transition_constraints =
             air.get_transition_constraints(&composition_coefficients.transition);
-
         // build periodic value table
         let periodic_values = PeriodicValueTable::new(air);
 
         // build boundary constraint groups; these will be used to evaluate and compose boundary
         // constraint evaluations.
-        let boundary_constraints =
-            BoundaryConstraints::new(air, &aux_rand_elements, &composition_coefficients.boundary);
+        let boundary_constraints = BoundaryConstraints::new(
+            air,
+            &aux_rand_elements,
+            &composition_coefficients.boundary,
+            composition_coefficients.lagrange.boundary,
+        );
+
+        let lagrange_constraints_evaluator = LagrangeConstraintsBatchEvaluator::new(
+            air,
+            aux_rand_elements.clone(),
+            composition_coefficients,
+        );
```

Comment:
I think using `aux_rand_elements` here is fine for now - but my current understanding is that the $r$-values for the Lagrange kernel column will be separate from these. But this is something we'll address in the next PR.